File: Backend/Microservices/ChatService/ChatService.Domain/Entities/ChatMessage.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace ChatService.Domain.Entities;

public class ChatMessage
{
  [BsonId]
  [BsonRepresentation(BsonType.ObjectId)]
  public string Id { get; set; } = ObjectId.GenerateNewId().ToString();

  public string SenderId { get; set; } = default!;

  public string ReceiverId { get; set; } = default!;

  public string Message { get; set; } = default!;

  public DateTime MessageTime { get; set; } = default!;
}

File: MobileApp/gymly/lib/models/chat_message.dart
class ChatMessage {
  final String id;
  final String senderId;
  final String receiverId;
  final String message;
  final DateTime messageTime;

  ChatMessage(
    this.id,
    this.senderId,
    this.receiverId,
    this.message,
    this.messageTime,
  );

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      json['id'] as String,
      json['senderId'] as String,
      json['receiverId'] as String,
      json['message'] as String,
      DateTime.parse(json["messageTime"]),
    );
  }
}
